packages/nx/src/plugins/js/lock-file/pnpm-parser.ts

```diff
@@ -759,17 +863,21 @@ function extractNameFromKey(key: string, isV5: boolean): string {
   if (key.startsWith('@')) {
     if (isV5) {
       const startFrom = key.indexOf('/');
-      return key.slice(0, key.indexOf('/', startFrom + 1));
+      const endIndex = key.indexOf('/', startFrom + 1);
+      return endIndex === -1 ? key : key.slice(0, endIndex);
     } else {
       // find the position of the '@'
-      return key.slice(0, key.indexOf('@', 1));
+      const atIndex = key.indexOf('@', 1);
+      return atIndex === -1 ? key : key.slice(0, atIndex);
```

Comment:
As a note, I updated this so that it works with "packageName" specifiers to return the original. I felt that made the helper a little more predictable and cleaned up my code a bit :)